coderd/userauth.go

```diff
@@ -636,19 +877,27 @@ func (api *API) oauthLogin(r *http.Request, params oauthLoginParams) (*http.Cook
 		return nil
 	}, nil)
 	if err != nil {
-		return nil, xerrors.Errorf("in tx: %w", err)
+		return nil, database.APIKey{}, xerrors.Errorf("in tx: %w", err)
 	}
 
-	cookie, err := api.createAPIKey(ctx, createAPIKeyParams{
+	cookie, key, err := api.createAPIKey(ctx, createAPIKeyParams{
 		UserID:     user.ID,
 		LoginType:  params.LoginType,
 		RemoteAddr: r.RemoteAddr,
 	})
 	if err != nil {
-		return nil, xerrors.Errorf("create API key: %w", err)
+		return nil, database.APIKey{}, xerrors.Errorf("create API key: %w", err)
 	}
 
-	return cookie, nil
+	return cookie, *key, nil
+}
+
+// setAnonUser passes a disposable user ID to force an audit diff
+// and generate a log for a failed login
+func setAnonUser(request *audit.Request[database.APIKey]) {
+	if request.New.UserID == uuid.Nil {
+		request.New = database.APIKey{UserID: uuid.New()}
```

Comment:
I have a worry that anyone programatically looking at these logs will not know if this UUID is an actual user or not. They might do an api call to `/users` to check.

I also don't know if a statis uuid is a good idea, eg `11111111-1111-1111-1111-111111111111`. 


Probably shouldn't gate this PR though. Just want to start some thoughts on it.. I will make an issue to track this idea.